### ISSUE TITLE
matplotlib & autogenes

### ISSUE THREAD

**Issue by joseagraz** [rnakato/docker_singlecell#11]
Hi
Your github repo is amazing!
I'm running into an issue where autogenes reads invalid metadata screwing up the matlib installation. Have any suggestions how to fix this problem?
Thanks
```text
172.0 Linking scanpy-1.9.8-pyhd8ed1ab_0
172.0 Linking scvi-tools-0.20.3-pyhd8ed1ab_1
172.0 Linking scvelo-0.2.5-pyhd8ed1ab_0
172.0 warning  libmamba [scvelo-0.2.5-pyhd8ed1ab_0] The following files were already present in the environment:
172.0     - lib/python3.9/site-packages/tests/__init__.py
172.0     - lib/python3.9/site-packages/tests/conftest.py
172.0 Linking cellrank-1.5.1-pyhdfd78af_0
172.2 
172.2 Transaction finished
172.2 
172.2 To activate this environment, use:
172.2 
172.2     micromamba activate shortcake_default
172.2 
172.2 Or to execute a single command in this environment, use:
172.2 
172.2     micromamba run -n shortcake_default mycommand
172.2 
172.2 
172.2 Installing pip packages: anndata2ri==1.3.1, autogenes==1.0.4, bbknn==1.6.0, cellmap==1.0.7, celltypist==1.6.2, dash==2.15.0, doubletdetection==4.2, gprofiler-official==1.0.0, magic-impute==3.0.0, liana==1.1.0, pyscenic==0.12.1, palantir==1.3.2, constclust==0.2.0, multivelo==0.1.3, optuna==3.5.0, ortools==9.7.*, pybind11==2.11.1, rpy2==3.5.15, screcode==0.2.5, scrublet==0.2.3, sctriangulate==0.13.0, snapatac2==2.4.0, velocyto==0.17.17, xlrd==2.0.1, session-info==1.0.0
172.8 Collecting anndata2ri==1.3.1 (from -r /opt/mambafR4yNRfRDJL (line 1))
172.8   Downloading anndata2ri-1.3.1-py3-none-any.whl.metadata (4.9 kB)
172.9 Collecting autogenes==1.0.4 (from -r /opt/mambafR4yNRfRDJL (line 2))
172.9   Downloading autogenes-1.0.4-py3-none-any.whl.metadata (2.8 kB)
173.0 WARNING: Ignoring version 1.0.4 of autogenes since it has invalid metadata:
173.0 Requested autogenes==1.0.4 from https://files.pythonhosted.org/packages/63/73/13814f04992e2af53d2f8339a5325611c37d372b83862ecbd858077d0218/autogenes-1.0.4-py3-none-any.whl (from -r /opt/mambafR4yNRfRDJL (line 2)) has invalid metadata: .* suffix can only be used with `==` or `!=` operators
173.0     matplotlib (>=3.0.*)
173.0                 ~~~~~~^
173.0 Please use pip<24.1 if you need to use this version.
173.0 ERROR: Could not find a version that satisfies the requirement autogenes==1.0.4 (from versions: 1.0, 1.0.1, 1.0.3, 1.0.4)
173.2 ERROR: No matching distribution found for autogenes==1.0.4
173.3 critical libmamba pip failed to install packages
------
failed to solve: process "/bin/bash -c eval \"$(micromamba shell hook --shell bash)\"     && micromamba create -y -n shortcake_default -f /opt/env.shortcake_default.yaml     && micromamba run -n shortcake_default python -m ipykernel install --name=shortcake_default     && micromamba run -n shortcake_default jupyter kernelspec list     && micromamba clean --all --yes     && rm /opt/env.shortcake_default.yaml" did not complete successfully: exit code: 1
```

**Comment by joseagraz** [comment]
-With pip versions > 24 this requirement specification is not valid anymore so installation fails. Only older pip versions allow the installation of autogenes at the moment.
-Found the wheel for autogenes with the syntax issue
Requires-Dist: matplotlib (>=3.0.*)
-Tried downgrading pip, but did not work for pip versions 23 & 22. pip below version 21 won't install anymore
-It would be ideal if we could build images using only local resources

**Comment by rnakato** [comment]
Hi joseagraz,
Thanks for the bug report about the autogenes installation. I'll have to fix it in the next update. Autogenes is available in ShortCake v3.0.0 and you can use it.
The Dockefiles are available in [Docker_Python](https://github.com/rnakato/ShortCake/tree/master/Docker_Python) directory, so you can modify them for your purpose.
